sample_size/metrics.py

```diff
@@ -0,0 +1,92 @@
+from typing import Union
+
+from statsmodels.stats.power import NormalIndPower
+from statsmodels.stats.power import TTestIndPower
+
+
+class BaseMetric:
```

Comment:
how about adding a constructor and set mde because they are the same across three types, and we can potentially also set `variance` and `default_power_analysis_instance` too
```python
class BaseMetric(object):

    __metaclass__ = ABCMeta

    def __init__(self, mde: float):
        self.mde = mde
    
    @property
    @abstractmethod
    def default_power_analysis_instance(self) -> Union[NormalIndPower, TTestIndPower]:
        raise NotImplementedError

    @property
    @abstractmethod
    def variance(self) -> float:
        raise NotImplementedError
    
    @staticmethod
    def check_positive(number: float, name: str) -> float:
        if number < 0:
            raise Exception(f"Please provide a positive number for {name}.")
        return number
```
And in each metric, we can do
```python
class NumericMetric(BaseMetric):
    def __init__(
        self,
        variance: float,
        mde: float,
    ):
        super(NumericMetric, self).__init__(mde)
        self._variance = self.check_positive(variance, "variance")

    @property
    def variance(self) -> float:
        return self._variance
    
    @property
    def default_power_analysis_instance(self) -> NormalIndPower:
        return TTestIndPower()
```